pokemongo_bot/cell_workers/evolve_all_worker.py

```diff
@@ -52,36 +52,47 @@ def work(self):
                 self._release_evolved(release_cand_list_ids)
 
     def _should_run(self):
-        # Will skip evolving if user wants to use an egg and there is none
-        skip_evolves = False
-
-        if self.config.evolve_all:
-            return skip_evolves
+        # Skip Evolving if there is no config
+        if not self.config.evolve_all or self.config.evolve_all is "NONE":
+            return False
 
         # Pop lucky egg before evolving to maximize xp gain
         use_lucky_egg = self.config.use_lucky_egg
         lucky_egg_count = self.bot.item_inventory_count(Item.ITEM_LUCKY_EGG.value)
 
-        if use_lucky_egg and lucky_egg_count > 0:
-            logger.log('Using lucky egg ... you have {}'
-                       .format(lucky_egg_count))
+        if not use_lucky_egg:
+            return True
+
+        # Try to use a lucky egg just at the beginning (for now)
+        if self.bot.tried_lucky_egg:
+            return True
```

Comment:
@douglascamata This one, right here, returns True after the bot first tried to use a lucky egg. This thing prevents the bot from using a lucky egg later and also spamming with messages.